astropy/nddata/decorators.py

```diff
@@ -175,16 +187,20 @@ def wrapper(data, *args, **kwargs):
                         # If it's in the func_args it's trivial but if it was
                         # in the func_kwargs we need to compare it to the
                         # default.
-                        # FIXME: This obviously fails if the explicitly given
-                        # value is identical to the default. No idea how that
-                        # could be achieved without replacing the signature or
-                        # parts of it. That would have other drawbacks like
-                        # making double-decorating nearly impossible and
-                        # violating the sphinx-documentation for that function.
+                        # Comparison to the default is done by comparing their
+                        # identity, this works because defaults in function
+                        # signatures are only created once and always reference
+                        # the same item.
+                        # FIXME: Python interns some values, for example the
+                        # integers from -5 to 255 (any maybe some other types
+                        # as well). In that case the default is
+                        # indistinguishable from an explicitly passed kwarg
+                        # and it won't notice that and use the attribute of the
+                        # NDData.
```

Comment:
They shouldn't but these are loose recommendations for `NDData`. `wcs` and `mask` are completly free of restrictions and could be anything.

I would feel better if the comment stays just so it's clear that it could be a problem, even if it's unlikely. But maybe the word `FIXME` could be dropped because I don't have any idea how to fix it 😄